From c3e99eebce4232909b5794e9cf201c3ce9cb4917 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Wed, 27 Jun 2012 10:47:05 +0200
Subject: [PATCH 02/10] override QT_USE_FILE to fix include order

this is a workaround for using QT_USE_FILE and having a header
called plugin.h which collides with same named header file from
mysql.

Cherry-picked from 5537e71
---
 soprano/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/soprano/CMakeLists.txt b/soprano/CMakeLists.txt
index 6389e29..28c3188 100644
--- a/soprano/CMakeLists.txt
+++ b/soprano/CMakeLists.txt
@@ -1,6 +1,8 @@
 project(soprano_core)
 
-include_directories(
+# overriding QT_USE_FILE since that includes QT_INCLUDE_DIR before soprano headers
+set(CMAKE_INCLUDE_DIRECTORIES_BEFORE ON)
+include_directories(BEFORE
   ${soprano_SOURCE_DIR}
   ${soprano_core_BINARY_DIR}
   ${soprano_core_SOURCE_DIR}
@@ -9,6 +11,7 @@ include_directories(
   ${soprano_core_SOURCE_DIR}/vocabulary
   ${QT_QTCORE_INCLUDE_DIR}
   ${QT_INCLUDE_DIR}
+  ${soprano_core_SOURCE_DIR}
   )
 
 configure_file(soprano-config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/soprano-config.h)
-- 
1.8.0.2

